compiler-cli/src/dependencies.rs

```diff
@@ -790,10 +803,16 @@ fn resolve_versions<Telem: Telemetry>(
     config: &PackageConfig,
     manifest: Option<&Manifest>,
     telemetry: &Telem,
+    packages_to_update: Option<&Vec<EcoString>>,
 ) -> Result<Manifest, Error> {
     telemetry.resolving_package_versions();
     let dependencies = config.dependencies_for(mode)?;
-    let locked = config.locked(manifest)?;
+    let mut locked = config.locked(manifest)?;
+
+    if let (Some(packages_to_update), Some(manifest)) = (packages_to_update, manifest) {
+        locked = unlock_packages(packages_to_update, &locked, Some(&manifest.packages))?;
+    }
+    let locked = locked.clone();
```

Comment:
Similar ownership things for this, can pass a mutable reference to `unlock_packages`